core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -1124,6 +1144,21 @@ object SnappyContext extends Logging {
     }
   }
 
+  private def failOnJarUnavailability(k: String, jars: Array[String], e: Throwable): Unit = {
+    GemFireXDUtils.waitForNodeInitialization()
+    Misc.getMemStore.getGlobalCmdRgn.remove(k)
+    if (!jars.isEmpty) {
+      val mutable = new ArrayBuffer[String]()
+      mutable += "__REMOVE_FILES_ONLY__"
+      jars.foreach(e => mutable += e)
+      RefreshMetadata.executeOnAll(globalSparkContext, RefreshMetadata.REMOVE_URIS_FROM_CLASSLOADER,
+        mutable.toArray)
+    }
```

Comment:
Question: Should we also catch and log exception here so that restart does not fail due to any exception here?